demos/burgers-goal_oriented.py

```diff
@@ -0,0 +1,425 @@
+
+        # set the first metric to the base
+        metric = metrics_i[0]
+        # all the other metrics
+        metric.average(*metrics_i[1:], weights=[dt] * len(metrics_i))
+
+        metrics.append(metric)
+
+    # Apply space time normalisation
+    space_time_normalise(metrics, mesh_seq.time_partition, mp)
+
+    # Adapt each mesh w.r.t. the corresponding metric, provided it hasn't converged
+    for i, metric in enumerate(metrics):
+        if not mesh_seq.converged[i]:
+            mesh_seq[i] = adapt(mesh_seq[i], metric)
+        complexities.append(metric.complexity())
+    num_dofs = mesh_seq.count_vertices()
+    num_elem = mesh_seq.count_elements()
+    pyrint(f"fixed point iteration {iteration + 1}:")
+    for i, (complexity, ndofs, nelem) in enumerate(
+        zip(complexities, num_dofs, num_elem)
+    ):
+        pyrint(
+            f"  subinterval {i}, complexity: {complexity:4.0f}"
+            f", dofs: {ndofs:4d}, elements: {nelem:4d}"
+        )
+
+    # Plot each intermediate adapted mesh
+    fig, axes = mesh_seq.plot()
+    for i, ax in enumerate(axes):
+        ax.set_title(f"Subinterval {i + 1}")
+    fig.savefig(f"burgers-anisotropic_mesh{iteration + 1}.jpg")
+    plt.close()
+
+    # Since we have two subintervals, we should check if the target complexity has been
+    # (approximately) reached on each subinterval
+    continue_unconditionally = np.array(complexities) < 0.90 * target
+    return continue_unconditionally
+
+
+# To avoid confusion, we redefine the :class:`GoalOrientedMeshSeq` object before using
+# it. ::
+
+mesh_seq = GoalOrientedMeshSeq(
+    time_partition,
+    meshes,
+    get_function_spaces=get_function_spaces,
+    get_initial_condition=get_initial_condition,
+    get_form=get_form,
+    get_solver=get_solver,
+    get_qoi=get_qoi,
+    qoi_type="time_integrated",
+)
+
+solutions = mesh_seq.fixed_point_iteration(
+    adaptor,
+    enrichment_kwargs={
+        "enrichment_method": "h",
+    },
+    parameters=params,
+)
+
+# .. code-block:: console
+#
+# fixed point iteration 1:
+#   subinterval 0, complexity:  457, dofs:  598, elements: 1110
+#   subinterval 1, complexity:  338, dofs:  442, elements:  799
+# fixed point iteration 2:
+#   subinterval 0, complexity:  681, dofs:  726, elements: 1361
+#   subinterval 1, complexity:  510, dofs:  564, elements: 1042
+# fixed point iteration 3:
+#   subinterval 0, complexity:  913, dofs:  971, elements: 1840
+#   subinterval 1, complexity:  676, dofs:  727, elements: 1350
+# QoI converged after 4 iterations under relative tolerance 0.001.
+
+
+# Finally, let's plot the adapted meshes. ::
+
+fig, axes = mesh_seq.plot()
+for i, ax in enumerate(axes):
+    ax.set_title(f"Subinterval {i + 1}")
+fig.savefig("burgers-anisotropic_mesh.jpg")
+plt.close()
+
+# .. figure:: burgers-anisotropic_mesh.jpg
+#    :figwidth: 100%
+#    :align: center
+#
+# This time, the elements are clearly anisotropic. This anisotropy is inherited from the
```

Comment:
Hmm are they "clearly" anisotropic? :) I don't see that big of a difference between the two methods